web/src/main/java/com/marklogic/hub/web/web/UtilController.java

```diff
@@ -16,46 +16,56 @@
  */
 package com.marklogic.hub.web.web;
 
+import com.marklogic.hub.impl.HubConfigImpl;
 import com.marklogic.hub.util.FileUtil;
 import com.marklogic.hub.web.model.SearchPathModel;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import org.springframework.stereotype.Controller;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestMethod;
-import org.springframework.web.bind.annotation.RequestParam;
-import org.springframework.web.bind.annotation.ResponseBody;
-
 import java.io.File;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
 
 @Controller
 @RequestMapping("/api/utils")
 public class UtilController {
 
     private static Logger logger = LoggerFactory.getLogger(UtilController.class);
 
+    @Autowired
+    private HubConfigImpl hubConfig;
+
 	@RequestMapping(value = "/searchPath", method = RequestMethod.GET)
 	@ResponseBody
 	public Map<String, Object> searchPath(@RequestParam String path) {
-		logger.debug("Search Path:" + path);
-		List<SearchPathModel> folders = new ArrayList<>();
-		List<SearchPathModel> files = new ArrayList<SearchPathModel>();
-		Path currentPath = Paths.get(".").toAbsolutePath().normalize();
-		Path relativePath;
+        String basePath = ".";
```

Comment:
We always had this behavior. Why are we changing this now ? During  bootstrapping, when the project is not initialized and hence when project dir not set , this may return null

Reply:
Check if hubprojectdir() is set before doing this, or else use that behavior.

Reply:
The behavior with this change seems to be 
1. When QS is started and hubConfig is not populated, the dir listing is relative to location from which war file was started (this can be seen in the dir picker in the login page)
2. Once logged in and hubProject/ hubConfig are populated, the dir listing is relative to current proj dir (this can be seen in dir picker in ingestion step page)
3. Once user logs out (old hubConfig/hubProject are still populated), when user gets back to login page and picks dir, it is relative to the last proj dir